src/main/java/net/minecraftforge/network/simple/IndexedMessageCodec.java

```diff
@@ -135,7 +135,7 @@ public <MSG> int build(MSG message, FriendlyByteBuf target)
     }
 
     void consume(FriendlyByteBuf payload, int payloadIndex, Supplier<NetworkEvent.Context> context) {
-        if (payload == null) {
+        if (payload == null || payload.readableBytes() == 0) {
```

Comment:
Perhaps `!payload.isReadable()` instead would be better?

Reply:
Wasn't aware that was a thing, but yeah seems logical, as then if there are other reasons it isn't readable it will catch them as well.